mcs/class/System/System.Net.Sockets/Socket.cs

```diff
@@ -982,14 +1012,14 @@ public IAsyncResult BeginAccept (int receiveSize, AsyncCallback callback, object
 			if (receiveSize < 0)
 				throw new ArgumentOutOfRangeException ("receiveSize", "receiveSize is less than zero");
 
-			SocketAsyncResult sockares = new SocketAsyncResult (this, state, callback, SocketOperation.AcceptReceive) {
+			SocketAsyncResult sockares = new SocketAsyncResult (this, callback, state, SocketOperation.AcceptReceive) {
```

Comment:
Because out of all the other implementation of `IAsyncResult`, `SocketAsyncResult` was the only one to have the state before the callback.